### PR TITLE
Add device name to pairing

### PR DESCRIPTION
**Pull request by cammellos**
Add device name to pairing section.

The first time a user access the devices section, they will be prompted to set a name for the device.

Once a name is set, it will be exposed to other devices when Pair this device is enabled (not through automatic discovery).

Also moves Devices section outside of advanced.
2019-01-14-154115_3840x1080_scrot

### Testing

1) Recover account A1 & A2
2) Enable development mode on both.
3) Go to settings -> Devices. You will be asked to set a name
4) Once a name is set, click on Pair this device (on both A1 & A2)
5) You should see A1 and A2 names in the list

In case we don't have this info (automatically discovered), "No Info (device-id)" will be shown. To see this you can recover A1 & A2, enable development mode, but don't pair. Send a 1-to-1 message to anyone from A1, A2 should now have A1 in the list.

Attached a screenshot @errorists what do you think?

status: ready

### CODE REVIEW

File: src/status_im/pairing/core.cljs [hunk 1]
```diff
@@ -243,16 +244,22 @@
...
-                          :last-paired     timestamp}]
+      (let [installation {:installation-id   installation-id
+                          :installation-name name
+                          :device-type       device-type
+                          :last-paired       timestamp}]
         (upsert-installation cofx installation)))))
 
+(fx/defn set-name [{:keys [db] :as cofx} installation-name]
+  (let [new-account (assoc (get-in cofx [:db :account/account]) :installation-name installation-name)]
+    {:db (assoc-in db [:account/account :installation-name] installation-name)
```
Comment (mandrigin) [inline src/status_im/pairing/core.cljs]: Why don’t we use `new-account` here? `(assoc-in db [:account/account] new-account)`

**Review by mandrigin — APPROVED** [review]
A small nitpick, looks overall good job!

File: src/status_im/pairing/core.cljs (hunk 1 above)
Comment (cammellos) [reply src/status_im/pairing/core.cljs]: yes, added that code later, best to use new-account, thanks

**Review by dmitryn — APPROVED** [review]

**Review by mandrigin — APPROVED** [review]

**Comment by mandrigin** [comment]
@churik it is "Done" but doesn't have neither "Tested-OK" nor "Tested-Issues"?

**Comment by churik** [comment]
Accidentally. Thanks for spotting!

**Comment by churik** [comment]
@cammellos to check actually sync we need  https://github.com/status-im/status-go/pull/1354 get merged?

**Comment by cammellos** [comment]
@churik no, this should not relay on that one, any issue?

**Comment by churik** [comment]
@cammellos 
After restarting app instead of custom names I see  "No Info (device-id)"  so devices aren't discovered automatically until you go to Devices > Pair and discover them again.

Steps:
- Recover account A1 & A2
- Enable development mode on both.
- Go to settings -> Devices. You will be asked to set a name
- Once a name is set, click on Pair this device (on both A1 & A2)
- You should see A1 and A2 names in the list. 
- Close app, wait ~1 min
- Open app, Go to settings -> Devices. Expect that can see device names. I fact: 
no

Logs:
[Status.log](https://github.com/status-im/status-react/files/2787788/Status.log)
[geth.log](https://github.com/status-im/status-react/files/2787797/geth.log)

Reproducible on all platforms.

**Comment by cammellos** [comment]
@churik thanks looks like I am not saving the name, I will take a look.

**Review by mandrigin — APPROVED** [review]

**Comment by cammellos** [comment]
@churik should be fixed now, thanks!

**Comment by churik** [comment]
Tested on IOS, Android, Mac OS, Windows - all devices can be synced on PR build. 
Also, retested pairing with 0.9.32-2 released build - working as expected, 'No info' is shown and devices are synced correctly

**Pull request merged**